### PR TITLE
feat(vpc): custom routing doc

### PR DESCRIPTION
**Pull request by RoRoJ**
Documentation for VPC custom routing feature.

### CODE REVIEW

File: network/vpc/how-to/manage-routing.mdx [hunk 1]
```diff
@@ -47,6 +39,19 @@ Routing is now activated as standard whenever you [create a new VPC](/network/vp
 
     Routing is activated on the VPC.
 
+## How to generate a managed route
+
+Two types of auto-generated routes exist:
+
+- **Local subnet route**: Generated when you create a Private Network in a VPC. Allows traffic to be routed between different Private Networks in the VPC.
+- **Default route to internet**: Generated when you attach a Public Gateway to a Private Network in the VPC, and set it to advertise a [default route](/network/public-gateways/concepts/#default-route). Allows traffic to be routed to addresses outside the VPC (i.e. the public internet) via the gateway.
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:47]:
```suggestion
- **Default route to internet**: Generated when you attach a Public Gateway to a Private Network in the VPC, and set it to advertize a [default route](/network/public-gateways/concepts/#default-route). Allows traffic to be routed to addresses outside the VPC (i.e. the public internet) via the gateway. 
```

File: network/vpc/how-to/manage-routing.mdx [hunk 2]
```diff
@@ -47,6 +39,19 @@ Routing is now activated as standard whenever you [create a new VPC](/network/vp
...
 
+## How to generate a managed route
+
+Two types of auto-generated routes exist:
+
+- **Local subnet route**: Generated when you create a Private Network in a VPC. Allows traffic to be routed between different Private Networks in the VPC.
+- **Default route to internet**: Generated when you attach a Public Gateway to a Private Network in the VPC, and set it to advertise a [default route](/network/public-gateways/concepts/#default-route). Allows traffic to be routed to addresses outside the VPC (i.e. the public internet) via the gateway. 
+
+<Message type="note">
+Public Gateways remain scoped to the Private Network(s) to which they are attached. They do not advertise the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:50]:
```suggestion
Public Gateways remain scoped to the Private Network(s) to which they are attached. They do not advertize the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 3]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
 
 3. Click the **Routing** tab.
 
-    If you have not yet created any Private Networks in the VPC, the route table will be empty. Routes are automatically generated and added to the route table when you either:
-    
+    Routes are automatically generated and added to the route table when you:
+
     - Create a Private Network in the VPC (this generates a **local subnet route**, which allows the VPC to automatically route traffic between Private Networks), or 
     - Attach a Public Gateway to a Private Network and set it to advertise a default route. This generates a **default route to the internet**.
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:68]:
```suggestion
    - Attach a Public Gateway to a Private Network and set it to advertize a default route. This generates a **default route to the internet**.
```

File: network/vpc/reference-content/understanding-routing.mdx [hunk 4]
```diff
@@ -34,26 +39,28 @@ The diagram below shows how routing works across two Private Networks on a VPC.
...
 
-A VPC's route table displays in the **Routing** tab of the console. The route table is populated with routes when you either create a Private Network in the VPC, or attach a Public Gateway to a Private Network and set it to advertise a default route. These routes are fully managed by Scaleway. Currently, only auto-created, managed routes are available, though in the future you will also be able to create your own custom routes.
+A VPC's route table displays in the **Routing** tab of the console. The route table is populated with routes when you either create a custom route,or create a Private Network in the VPC, or attach a Public Gateway to a Private Network and set it to advertise a default route.
```
Comment (nerda-codes) [inline network/vpc/reference-content/understanding-routing.mdx:56]:
```suggestion
A VPC's route table displays in the **Routing** tab of the console. The route table is populated with routes when you either create a custom route, or create a Private Network in the VPC, or attach a Public Gateway to a Private Network and set it to advertize a default route.
```

**Review by nerda-codes — COMMENTED** [review]
not sure about the "advertize" suggestions, ffti

File: network/vpc/concepts.mdx [hunk 5]
```diff
@@ -89,13 +89,18 @@ While DHCP is built into all new Private Networks, it may not be automatically a
 
 A [route table](/network/vpc/reference-content/understanding-routing/#route-table) is used to manage and control the routing of traffic within a VPC. The routes within a route table tell the VPC where to send traffic trying to get to a specific destination IP address. One line in the route table corresponds to one route.
 
-Routes can either be of type `local subnet route` for routes to Private Networks, or `default route to internet` for routes to Public Gateways. When deciding which route to apply, the route table reads the routes from most specific to least specific, in terms of destination IP range. The first matching route encountered is the one that determines the path for the traffic. Therefore, a route to destination `172.16.8.0/22` is applied before a default route to `0.0.0.0/0`.
+Routes can be of the following types:
+- Auto-generated, fully managed `Local subnet routes` for routes to Private Networks
+- Auto-generated, fully managed `Ddefault route to internet` for routes to Public Gateways.
```
Comment (ldecarvalho-doc) [inline network/vpc/concepts.mdx:94]:
```suggestion
- Auto-generated, fully managed `Default route to internet` for routes to Public Gateways.
```

**Review by bene2k1 — APPROVED** [review]

File: network/vpc/concepts.mdx [hunk 6]
```diff
@@ -89,13 +89,18 @@ While DHCP is built into all new Private Networks, it may not be automatically a
 
 A [route table](/network/vpc/reference-content/understanding-routing/#route-table) is used to manage and control the routing of traffic within a VPC. The routes within a route table tell the VPC where to send traffic trying to get to a specific destination IP address. One line in the route table corresponds to one route.
 
-Routes can either be of type `local subnet route` for routes to Private Networks, or `default route to internet` for routes to Public Gateways. When deciding which route to apply, the route table reads the routes from most specific to least specific, in terms of destination IP range. The first matching route encountered is the one that determines the path for the traffic. Therefore, a route to destination `172.16.8.0/22` is applied before a default route to `0.0.0.0/0`.
+Routes can be of the following types:
+- Auto-generated, fully managed `Local subnet routes` for routes to Private Networks
```
Comment (jcirinosclwy) [inline network/vpc/concepts.mdx:93]:
```suggestion
- Auto-generated, fully managed `Local subnet routes` for routes to Private Networks.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 7]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
-
-    As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertising the default route, these routes will be automatically deleted from your route table.
+    For help with understanding the route table and how to read it, [refer to our documentation about route tables](/network/vpc/reference-content/understanding-routing/#route-table).
 
 ### How to view VPC routes in IPV6 
 
-Scaleway VPC routing supports both IPv4 and IPv6 protocols. Managed routes to Private Networks are simultaneously generated for both IPV4 and IPV6, and both are added to the route table. Use the toggle above the route table to switch from the default view of **IPV4** routes, to a view of **IPV6** routes.
\ No newline at end of file
+Scaleway VPC routing supports both IPv4 and IPv6 protocols. Managed routes to Private Networks are simultaneously generated for both IPV4 and IPV6, and both are added to the route table. Use the toggle above the route table to switch from the default view of **IPV4** routes, to a view of **IPV6** routes.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:79]:
```suggestion
Scaleway VPC routing supports both IPv4 and IPv6 protocols. Managed routes to Private Networks are simultaneously generated for both IPV4 and IPV6, and both are added to the route table. Use the toggle above the route table to switch from the default view of **IPV4** routes to a view of **IPV6** routes.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 8]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+    <Lightbox src="scaleway-create-custom-route.webp" alt="The Scaleway console Custom Route wizard displays, with the sections detailed in the text below."/>
+
+5. Enter a **description** for your route, e.g. "Route to hosted VPN".
+
+6. Optionally, enter one or more **tags** for your route. Tags help you filter and organize your list of routes.
+
+7. Enter a **destination** for the route. The VPC will apply the route to all traffic with a matching destination IP. You must enter an IPv4 or IPv6 CIDR range with a subnet mask, e.g. `192.168.1.0/24`. For a single IP address, use the `/32` mask for IPv4.
+
+8. Enter a **next hop** for the route. The VPC will route traffic for the destination IP to the resource designated as next hop.
+    - Select a resource type: **Instance**, **Public Gateway** or **Elastic Metal**. Routing is not yet compatible with Managed Databases, nor with other types of Scaleway resource which are not integrated with VPC.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:108]:
```suggestion
    - Select a resource type: **Instance**, **Public Gateway** or **Elastic Metal**. Routing is not yet compatible with Managed Databases, nor with other types of Scaleway resources which are not integrated with VPC. 
```

File: network/vpc/how-to/manage-routing.mdx [hunk 9]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+4. Click the delete <Icon name="delete" /> icon next to the custom route you want to delete. A pop-up displays asking you to confirm.
+
+5. Click **Delete route** to confirm.
+
+    The custom route is deleted, and you are returned to the list of your VPC's routes.
+
+
+
+
+
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:173]:
```suggestion
```

File: network/vpc/how-to/manage-routing.mdx [hunk 10]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+4. Click the delete <Icon name="delete" /> icon next to the custom route you want to delete. A pop-up displays asking you to confirm.
+
+5. Click **Delete route** to confirm.
+
+    The custom route is deleted, and you are returned to the list of your VPC's routes.
+
+
+
+
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:172]:
```suggestion
```

File: network/vpc/how-to/manage-routing.mdx [hunk 11]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+3. Click the **Routing** tab.
+
+4. Click the delete <Icon name="delete" /> icon next to the custom route you want to delete. A pop-up displays asking you to confirm.
+
+5. Click **Delete route** to confirm.
+
+    The custom route is deleted, and you are returned to the list of your VPC's routes.
+
+
+
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:171]:
```suggestion
```

File: network/vpc/how-to/manage-routing.mdx [hunk 12]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+3. Click the **Routing** tab.
+
+4. Click the delete <Icon name="delete" /> icon next to the custom route you want to delete. A pop-up displays asking you to confirm.
+
+5. Click **Delete route** to confirm.
+
+    The custom route is deleted, and you are returned to the list of your VPC's routes.
+
+
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:170]:
```suggestion
```

File: network/vpc/how-to/manage-routing.mdx [hunk 13]
```diff
@@ -57,19 +62,112 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+2. Click a VPC on which you have activated routing.
+
+3. Click the **Routing** tab.
+
+4. Click the delete <Icon name="delete" /> icon next to the custom route you want to delete. A pop-up displays asking you to confirm.
+
+5. Click **Delete route** to confirm.
+
+    The custom route is deleted, and you are returned to the list of your VPC's routes.
+
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:169]:
```suggestion
```

**Review by jcirinosclwy — APPROVED** [review]

File: network/vpc/how-to/manage-routing.mdx [hunk 14]
```diff
@@ -57,19 +62,106 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+    <Lightbox src="scaleway-create-custom-route.webp" alt="The Scaleway console Custom Route wizard displays, with the sections detailed in the text below."/>
+
+5. Enter a **description** for your route, e.g. "Route to hosted VPN".
+
+6. Optionally, enter one or more **tags** for your route. Tags help you filter and organize your list of routes.
+
+7. Enter a **destination** for the route. The VPC will apply the route to all traffic with a matching destination IP. You must enter an IPv4 or IPv6 CIDR range with a subnet mask, e.g. `192.168.1.0/24`. For a single IP address, use the `/32` mask for IPv4.
+
+8. Enter a **next hop** for the route. The VPC will route traffic for the destination IP to the resource designated as next hop.
+    - Select a resource type: **Instance**, **Public Gateway** or **Elastic Metal**. Routing is not yet compatible with Managed Databases, nor with other types of Scaleway resources which are not integrated with VPC.
```
Comment (RoRoJ) [inline network/vpc/how-to/manage-routing.mdx:108]: Check if this is now preceded by PN selection

File: network/vpc/how-to/manage-routing.mdx [hunk 15]
```diff
@@ -57,19 +62,106 @@ Your VPC's **route table** can be found in its **Routing** tab. The route table
...
+
+8. Enter a **next hop** for the route. The VPC will route traffic for the destination IP to the resource designated as next hop.
+    - Select a resource type: **Instance**, **Public Gateway** or **Elastic Metal**. Routing is not yet compatible with Managed Databases, nor with other types of Scaleway resources which are not integrated with VPC. 
+    - Select the **name** of the specific resource you want to route traffic to. The resource must be attached to a Private Network in this VPC.
+
+    <Message type="note">
+    If you entered an IPv6 destination, resources not compatible with IPv6, e.g. Public Gateways, will not be available for selection as next hop.
+    </Message>
+
+    If the selected resource is attached to multiple Private Networks within the VPC, you are prompted to select the Private Network that should be used for the route.
```
Comment (RoRoJ) [inline network/vpc/how-to/manage-routing.mdx:115]: Potentially remove

File: network/vpc/concepts.mdx [hunk 16]
```diff
@@ -89,13 +89,18 @@ While DHCP is built into all new Private Networks, it may not be automatically a
...
+Routes can be of the following types:
+- Auto-generated, fully managed `Local subnet routes` for routes to Private Networks.
+- Auto-generated, fully managed `Default route to internet` for routes to Public Gateways.
+- Custom routes created by the user.
+
+When deciding which route to apply, the route table reads the routes from most specific to least specific, in terms of destination IP range. The first matching route encountered is the one that determines the path for the traffic. Therefore, a route to destination `172.16.8.0/22` is applied before a default route to `0.0.0.0/0`.
 
 ## Routing
 
 Routing allows Private Networks in the same VPC to communicate with each other, via automatically-created and managed routes. Routing is activated by default whenever you create a new VPC, and can be activated on pre-existing VPCs by [following these steps](/network/vpc/how-to/manage-routing/#how-to-activate-routing-on-an-existing-vpc).
```
Comment (RoRoJ) [inline network/vpc/concepts.mdx:101]:
```suggestion
Routing allows Private Networks in the same VPC to communicate with each other, via both automatically-created/managed and custom routes. Routing is activated by default whenever you create a new VPC, and can be activated on pre-existing VPCs by [following these steps](/network/vpc/how-to/manage-routing/#how-to-activate-routing-on-an-existing-vpc).
```

**Pull request merged**